File: 137.py
#!/usr/bin/env python

"""This script solves the Project Euler problem "Fibonacci golden nuggets".
The problem is: Find the 15th golden nugget.
"""

import argparse
import math


def main(args):
    """Fibonacci golden nuggets"""

    print(fib(2 * args.ordinal) * fib(2 * args.ordinal + 1))


def fib(n):
    """Calculate Fibonacci number"""

    return int((pow(1 + math.sqrt(5), n) - pow(1 - math.sqrt(5), n)) /
               (pow(2, n) * math.sqrt(5)))

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Fibonacci golden nuggets')
    parser.add_argument(
        'ordinal', metavar='ORDINAL', type=int, default=15, nargs='?',
        help='The required ordinal golden nugget')
    args = parser.parse_args()

    main(args)

File: 137.js
#!/usr/bin/env node

// Fibonacci golden nuggets

var program = require('commander');

program
    .version('0.1.0')
    .description('Fibonacci golden nuggets')
    .option('-o, --ordinal <int>',
        'The required ordinal golden nugget', Number, 15)
    .parse(process.argv);

console.log(fib(2 * program.ordinal) * fib(2 * program.ordinal + 1));

function fib(n) {
    return parseInt((Math.pow(1 + Math.sqrt(5), n) -
            Math.pow(1 - Math.sqrt(5), n)) /
        (Math.pow(2, n) * Math.sqrt(5)) + 0.5);
}
